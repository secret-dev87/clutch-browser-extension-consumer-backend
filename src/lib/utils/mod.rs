use ethers::types::H160;

pub fn convert_to_hex(addr: H160) -> String {
    format!("0x{}", hex::encode(addr))
}
