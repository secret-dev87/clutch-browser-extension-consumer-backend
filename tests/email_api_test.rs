use axum::http::StatusCode;
use axum_test_helper::{TestClient, TestResponse};
use lib::test::utils::setup;
use lib::test::utils::tear_down;

#[tokio::test]
async fn test_verify_email() {
    let (client, _app_state, db_url) = setup().await;

    let res = client
        .post("/email/verify")
        .body("{\"email\":\"a@me.com\"}")
        .header("Content-Type", "application/json")
        .send()
        .await;
    assert_eq!(res.status(), StatusCode::OK);
    insta::assert_yaml_snapshot!(res.text().await);

    tear_down(db_url).await;
}

#[tokio::test]
async fn test_invalid_email_format() {
    let (client, _app_state, db_url) = setup().await;

    let res = client
        .post("/email/verify")
        .body("{\"email\":\"a\"}")
        .header("Content-Type", "application/json")
        .send()
        .await;
    assert_eq!(res.status(), StatusCode::OK);
    insta::assert_yaml_snapshot!(res.text().await);

    tear_down(db_url).await;
}

#[tokio::test]
async fn test_error_when_exceeds_email_send_limit() {
    let (client, _app_state, db_url) = setup().await;

    async fn verify_email(client: &TestClient) -> TestResponse {
        client
            .post("/email/verify")
            .body("{\"email\":\"exceeds@example.com\"}")
            .header("Content-Type", "application/json")
            .send()
            .await
    }

    for _ in 0..5 {
        verify_email(&client).await;
    }

    let res = verify_email(&client).await;
    assert_eq!(res.status(), StatusCode::OK);
    insta::assert_yaml_snapshot!(res.text().await);

    tear_down(db_url).await;
}
