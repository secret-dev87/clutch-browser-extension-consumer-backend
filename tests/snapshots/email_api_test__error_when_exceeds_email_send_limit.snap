---
source: tests/email_api_test.rs
expression: res.text().await
---
"{\"status\":\"Error\",\"payload\":{\"Error\":{\"error_message\":\"Email verification limit exceeded (attempts: 5, limit: 4) for email: exceeds@example.com\"}}}"

